schema_enforcer/schemas/validator.py

```diff
@@ -84,27 +86,110 @@ def validate(self, data: dict, strict: bool):  # pylint: disable=W0613
             self.add_validation_error(self.error)
 
 
+class PydanticValidation(BaseValidation):
+    """Basic wrapper for Pydantic models to be used as validators."""
+
+    model: BaseModel
+
+    def validate(self, data: dict, strict: bool = False):
+        """Validate data against Pydantic model.
+
+        Args:
+            data (dict): variables to be validated by validator
+            strict (bool): true when --strict cli option is used to request strict validation (if provided)
+
+        Returns:
+            None
+
+        Use add_validation_error and add_validation_pass to report results.
+        """
+        try:
+            self.model.model_validate(data, strict=strict)
+            self.add_validation_pass()
+        except ValidationError as err:
+            self.add_validation_error(str(err))
+
+
 def is_validator(obj) -> bool:
     """Returns True if the object is a BaseValidation or JmesPathModelValidation subclass."""
     try:
-        return issubclass(obj, BaseValidation) and obj not in (JmesPathModelValidation, BaseValidation)
+        return (issubclass(obj, BaseValidation) or issubclass(obj, BaseModel)) and obj not in (
+            BaseModel,
+            BaseValidation,
+            JmesPathModelValidation,
+        )
     except TypeError:
         return False
 
 
-def load_validators(validator_path: str) -> dict[str, BaseValidation]:
-    """Load all validator plugins from validator_path."""
+def pydantic_validation_factory(orig_model) -> PydanticValidation:
+    """Create a PydanticValidation instance from a Pydantic model."""
+    return type(
+        orig_model.__name__,
+        (PydanticValidation,),
+        {
+            "id": f"{orig_model.id}",
+            "top_level_properties": set([property for property in orig_model.model_fields]),
+            "model": orig_model,
+        },
+    )
+
+
+def load_pydantic_validators(
+    model_packages: list,
+) -> dict[str, PydanticValidation]:
+    """Load all validator plugins from validator_packages."""
     validators = {}
-    for importer, module_name, _ in pkgutil.iter_modules([validator_path]):
+    for package in model_packages:
+        module_name, attr = package.split(":")
+        module = __import__(module_name, fromlist=[attr])
+        manager = getattr(module, attr)
```

Comment:
Maybe add some error handling here? Or are ImportError and AttributeError sufficiently clear if/when they occur?